SharedLibs/TileMapHelper/src/model/TileMapModel.ts

```diff
@@ -330,12 +436,58 @@ export class EditableTileMapLayer extends AbstractEditableLayer {
    */
   constructor(tileMap: EditableTileMap, id: integer) {
     super(tileMap, id);
+    this.buildEmptyLayer(
+      this.tileMap.getDimensionX(),
+      this.tileMap.getDimensionY()
+    );
+    this._alpha = 1;
+  }
+
+  buildEmptyLayer(dimensionX: number, dimensionY: number) {
     this._tiles = [];
-    this._tiles.length = this.tileMap.getDimensionY();
+    this._tiles.length = dimensionY;
     for (let index = 0; index < this._tiles.length; index++) {
-      this._tiles[index] = new Int32Array(this.tileMap.getDimensionX());
+      this._tiles[index] = new Int32Array(dimensionX);
     }
-    this._alpha = 1;
+  }
+
+  static from(
+    editableTileMapLayerAsJsObject: any,
```

Comment:
Should this be typed?

Reply:
I'm not sure. We should leave it as `any` if we consider the data can be corrupt. Or type it if we're confident the data will have the right structure

Reply:
Why would it be corrupted?

Reply:
I don't know 😅 I'm adding the types then